### DISCUSSION TITLE
How to use it with a standalone AppComponent or a Web Component

### DISCUSSION THREAD

**Discussion by YeaTiii** [ocombe/ng2-translate#1474] [Q&A]
In Angular 17, componenents are standalone by default, in the documentation, ngx-translate is loaded thru the module, how can I load it if I don't have a module ?
Also, when developping Web Components (@angular/elements), how can I use it too ?

**Comment by ph1823** [comment]
I think this comment can help you: https://github.com/ngx-translate/core/issues/1460#issuecomment-1837600916

**Comment by cuddlecake** [comment]
Modules, such as NgxModule, are a collection of components and providers. Static module methods such as `forRoot` are specifically just for setting different providers to initialize application / environment.

The general method for retrieving providers from a Module is [importProvidersFrom](https://angular.io/api/core/importProvidersFrom)

E.g.
```ts
bootstrapApplication(AppComponent, { providers: [importProvidersFrom(TranslateModule.forRoot({ /* config */ }))] })
```

**Comment by rbalet** [comment]
Please, read the latest documentation, you should find you answer there: https://ngx-translate.org/getting-started/translating-your-components/
